src/diffusers/pipelines/unidiffuser/modeling_text_decoder.py

```diff
@@ -0,0 +1,220 @@
+from typing import Optional
+
+import numpy as np
+import torch
+from torch import nn
+from transformers import GPT2Config, GPT2LMHeadModel
+from transformers.modeling_utils import ModuleUtilsMixin
+
+from ...configuration_utils import ConfigMixin, register_to_config
+from ...models import ModelMixin
+
+
+# Modified from ClipCaptionModel in https://github.com/thu-ml/unidiffuser/blob/main/libs/caption_decoder.py
+class UniDiffuserTextDecoder(ModelMixin, ConfigMixin, ModuleUtilsMixin):
```

Comment:
Should there be a `from_pretrained()` class method for this to handle weight loading?

Reply:
I think my assumption here was that the `from_pretrained()` method inherited from `ModelMixin` would suffice - would we instead need to override it?

Note that I followed the design from here: https://github.com/huggingface/diffusers/blob/a9477bbdac9b8b1755e0383e5198e4baa62678b1/src/diffusers/pipelines/spectrogram_diffusion/continous_encoder.py#L29 and it looks like they don't override the `from_pretrained()` method there (see https://github.com/huggingface/diffusers/pull/2963#discussion_r1159535224 for reference).

Reply:
Yeah, you're correct.